### ISSUE TITLE
The animation function give animation reversed, the update function is not correct (I think)

### ISSUE THREAD

**Issue by abdo71999** [einsteinpy/einsteinpy#655]
The animate function in StaticGeodesicPlotter gives animations reversed

📋  **Steps to solve the problem**

I am not programming savvy at all I am doing a presentation on the EinsteinPy package I tried to create some animations and they gave a gif that is reversed, so I looked at the source code and with the help of ChatGPT I found that the update function is the problem. 
Location: einsteinpy\plotting\geodesic\static.py
![image](https://github.com/einsteinpy/einsteinpy/assets/121961749/a0aecba7-5546-4775-9fcd-913c11f6b0d0)
The old update function: 

        def _update(num, data, line):
            line.set_data(data[:2, :num])
            line.set_3d_properties(data[2, :num])

            return (line,)
The new update function that works (for me):

        def _update(num, data, line):
            current_frame = num % N  # Ensure animation plays in the correct order
            reversed_frame = N - current_frame - 1  # Reverse the frame order
            line.set_data(data[:2, :reversed_frame + 1])
            line.set_3d_properties(data[2, :reversed_frame + 1])

            return (line,)

**Comment by JeS24** [comment]
@abdo71999 Thanks for opening the issue. I'll take a look today.

**Comment by JeS24** [comment]
@abdo71999 Can you share the code snippet for which the animation was reversed? Looking at the code itself, the indexing and update functions are correct as-is. There might be some problem at the time of getting the geodesic solve.

**Comment by abdo71999** [comment]
Here it is. It was a part of a presentation I did on the package I did it
as an example. I am using VScode so I used "%matplotlib ipympl"
much regards

**Comment by JeS24** [comment]
@abdo71999 It seems you wanted to attach some code? It's not appearing on my side though.
![image](https://github.com/einsteinpy/einsteinpy/assets/31299799/3557d11a-48e9-4ad5-a6ae-9a5d7da6fb2a)

Did you use this code - https://docs.einsteinpy.org/en/latest/examples/Animations%20in%20EinsteinPy.html ?

**Comment by abdo71999** [comment]
No, I was trying to attach a Jupyter Notebook and 2 GIF  files for the animations before and after the new update function. BTW I discovered that the new update function reverses other animations so there is still a problem

![KerrNewmann(with new update function)](https://github.com/einsteinpy/einsteinpy/assets/121961749/ca42f179-a7d6-4d9c-9919-a228b5ed91fe)
![KerrNewmann](https://github.com/einsteinpy/einsteinpy/assets/121961749/baa436f3-9b84-45b0-86f2-caa92a1e66f5)
[example on reversed animation.docx](https://github.com/einsteinpy/einsteinpy/files/14079914/example.on.reversed.animation.docx)

**Comment by JeS24** [comment]
Thanks for the plots & the code. I think, the update function is okay as it is updating the plot starting from the first 3D coordinate in the geodesic to the last. However, ...
> BTW I discovered that the new update function reverses other animations so there is still a problem

...if this is true, then it is more likely to be an inverted sign somewhere in the definition of (some of) the metric functions (`einsteinpy/src/einsteinpy/geodesic/utils.py`). Let me use another solver and another test case to confirm that the results are indeed reversed. Will update here in some time.

**Comment by JeS24** [comment]
Preliminary tests show that it is a sign issue in the calculation of the 4-momentum. Will check the associated code soon.
